chain/ethereum/src/ethereum_adapter.rs

```diff
@@ -1499,6 +1504,22 @@ pub(crate) async fn blocks_with_triggers(
         .compat()
         .await?;
 
+    // Filter out call triggers that come from unsuccessful transactions
+
+    let mut blocks =
+        if unified_api_version.equal_or_greater_than(&graph::data::subgraph::API_VERSION_0_0_5) {
+            let section = stopwatch_metrics
+                .start_section("filter_call_triggers_from_unsuccessful_transactions");
+            let futures = blocks.into_iter().map(|block| {
+                filter_call_triggers_from_unsuccessful_transactions(block, &eth, &chain_store)
+            });
+            let blocks = futures03::future::try_join_all(futures).await?;
+            section.end();
+            blocks
+        } else {
+            blocks
+        };
+
```

Comment:
Not sure if this is the most performant way to do `async/await`, but I opted to do it the simplest way without making calls wait for each other.  

Also, this `let/if/else` is necessary due to `filter_call_triggers_from_unsuccessful_transactions` consuming and returning the `blocks` value. 

I tried to go with `&mut` but I found a lot of borrow-checking problems. Maybe there is a better way to do it.